command/agent/agent_test.go

```diff
@@ -1060,6 +1060,109 @@ func TestServer_ShouldReload_ShouldHandleMultipleChanges(t *testing.T) {
 	}
 }
 
+func TestServer_ShouldReload_ReturnTrueForVaultConfigChanges(t *testing.T) {
+	t.Parallel()
+	require := require.New(t)
+
+	agent := &Agent{
+		logger: log.New(ioutil.Discard, "", 0),
+		config: DevConfig(),
+	}
+
+	vaultConfig := &sconfig.VaultConfig{
+		Enabled: helper.BoolToPtr(true),
+		Addr:    "http://127.0.0.1:8200",
+		Role:    "nomad-cluster",
+		Token:   "4f7cece4-5035-f5e6-ba8f-18314a7a0333",
+	}
+	agent.config.Vault = vaultConfig
+
+	newConfig := DevConfig()
+	newVaultConfig := &sconfig.VaultConfig{
+		Enabled: helper.BoolToPtr(true),
+		Addr:    "http://127.0.0.1:8200",
+		Role:    "nomad-cluster",
+		Token:   "7c17ef0b-83df-b6f0-5401-d419",
+	}
+	newConfig.Vault = newVaultConfig
+
+	shouldReloadAgent, shouldReloadHTTP, shouldReloadRPC := agent.ShouldReload(newConfig)
+	require.True(shouldReloadAgent)
+	require.False(shouldReloadHTTP)
+	require.True(shouldReloadRPC)
+}
+
+func TestServer_ShouldReload_ForVaultConfigUpgradeOrDowngrade(t *testing.T) {
+	t.Parallel()
+	require := require.New(t)
+
+	// Test downgrading a vault configuration
```

Comment:
Suggest making this a table driven test to make it easy to add more, and read clarity